plugins/common/opcua/input/input_client.go

```diff
@@ -0,0 +1,371 @@
+package input
+
+import (
+	"context"
+	"fmt"
+	"github.com/gopcua/opcua/ua"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/common/opcua"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type TimestampSource string
+
+const (
+	TimestampSourceServer   TimestampSource = "server"
+	TimestampSourceSource   TimestampSource = "source"
+	TimestampSourceTelegraf TimestampSource = "gather"
+)
+
+// NodeSettings describes how to map from a OPC UA node to a Metric
+type NodeSettings struct {
+	FieldName      string     `toml:"name"`
+	Namespace      string     `toml:"namespace"`
+	IdentifierType string     `toml:"identifier_type"`
+	Identifier     string     `toml:"identifier"`
+	DataType       string     `toml:"data_type"`   // Kept for backward compatibility but was never used.
+	Description    string     `toml:"description"` // Kept for backward compatibility but was never used.
+	TagsSlice      [][]string `toml:"tags"`
```

Comment:
Tags are usually a `map[string]string` within telegraf config, so they can be configured as such:
```toml
  tag1 = "value1"
  tag2 = "value2"
```
IMHO more clear than using arrays with 2 items.

Reply:
I definitely agree, however this is how it was done before, so this would not be backward compatible.

Reply:
You are correct.

However it could be done by adding an additional field like for example `default_tags`  and deprecate the old `tags` section.

Reply:
I have created the new property `default_tags`. If any are defined, the `tags` slices are ignored.